README.md

```diff
@@ -1,100 +1,139 @@
+
+TODO: Delete this and the text above, and describe your gem
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'appdev_support', git: "https://github.com/firstdraft/appdev_support"
+```
+
+And then execute:
+
+    $ bundle install
+
+Or install it yourself as:
+
+    $ gem install appdev_support
+
+See [iniliazer](#configuration).
+
+## Usage
+
+Once installed, you can do things like:
+
+```ruby
+# in app/controllers/events_controller.rb
+class EventsController < ApplicationController 
+  def index
+    cookies.store(:theme, "dark")
+    cookies.fetch(:theme)
+    session.fetch(:user_id)
+  end
+end
+```
+
+and `session.fetch` will raise a `KeyNotFound` Exception.
+
+`.fetch` can be called on `session`, `cookies`, and any `ActiveRecord` object. For an `ActiveRecord` object, you can `.fetch` a column.
+
+```ruby
+Event.fetch(:starts_on) # => "Mon, 23 Mar 2020 23:25:48"
+```
+
+Similarly, `.store` can be called on `session` and `cookies` with the expected behavior. 
+
+```ruby
+session.store(:user_id, 1)
+```
+
+Displaying an `ActiveRecord:Relation` in a View file will display:
+
+```html
+ActiveRecord:Relation:Event (array with 4 Event instances inside)
+```
+
+You can call `.at` on an `ActiveRecord:Relation` instead of just `[]` to mirror how Arrays work.
+
+```ruby
+Events.all.at(0)
+```
+
+If the containing app has `pry` or `pry-rails` installed, the `print` functionality has been enhanced:
+With the value `:base`:
+- `ActiveRecord::Relation` objects are displayed like this:
+  ```irb
+  pry(main)> Todo.all
+  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
+  ```
+With the value `:full`
+- `ActiveRecord::Relation` objects are displayed like this:
+  ```irb
+  pry(main)> Todo.all
+  Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
+  Todo Load (0.2ms)  SELECT "todos".* FROM "todos"
+  +----+-------+------+---------------------------------------------+
+  | id | title | body | {:header_frequency=>10, :border=>:markdown} |
+  +----+-------+------+---------------------------------------------+
+  |  1 | test  | nope |                                             |
+  +----+-------+------+---------------------------------------------+
+
+    Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
+  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
+  ```
+- `Class` objects are displayed like this:
+  ```irb
+  pry(main)> Todo
+  => Todo(id: integer, title: string, body: text, created_at: datetime, updated_at: datetime)
+
+  The Todo class itself.
+  ```
+- All other objects are displayed like this:
+  ```irb
+  pry(main)> "hello, world"
+  => "hello, world"
+
+  An instance of the String class.
+  ```
+
+## Configuration
+
+Add an initializer:

```

Comment:
It took me a few to understand how this is setup. Partly just due to me being new to the dirty details of Rails configuration. But maybe you want to be explicit here and say (this is how I set it up, maybe there is a more correct way)
```suggestion
In your Rails application, add an initializer `config/initializers/appdev_support.rb`, and put the following code in it:
```

Reply:
Perhaps a middle path: something like,

"Add an initializer to your application that contains the following:

...

In Rails, that would mean creating a file in the `config/` folder called e.g. `appdev_support.rb`."